x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlMetadata.java

```diff
@@ -224,7 +224,7 @@ public MlMetadataDiff(StreamInput in) throws IOException {
         public void writeTo(StreamOutput out) throws IOException {
             jobs.writeTo(out);
             datafeeds.writeTo(out);
-            if (out.getVersion().onOrAfter(Version.V_7_0_0)) {
+            if (out.getVersion().onOrAfter(Version.V_6_7_0)) {
```

Comment:
would it be possible to fix this separately next time, and add unit tests for serialization that fail with the wrong version conditionals?